### PR TITLE
Log swallowed HttpRequestExceptions

### PR DESCRIPTION
**Pull request by Manolachi**
## Type of change
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective
This PR addresses the issue raised in https://github.com/bitwarden/server/issues/1093. Since the HttpRequestExceptions was swallowed, users couldn't figure out what went wrong. By logging the issue, further investigations can now be made, to identify the culprit.

Additionally, the commit updates the internal web vault URLs used in Development mode, as they were using http and an outdated port number which is no longer used.

## Code changes
* **Admin/Controllers/HomeController.cs:** Added a logger which is injected through the constructor and used when an HttpRequestException is thrown to reveal what went wrong when sending the GET request to the specified Uri.
* **appsettings.Development.json:** Updated this file in every project to point to the correct URL of the web vault.

## Testing requirements
Verify that the errors that occur when sending a request to dockerhub or the web vault for retrieving the product version by accessing the Home page of the Bitwarden Admin Portal are successfully logged.

## Before you submit
- [x] I have checked for formatting errors (`dotnet tool run dotnet-format --check`) (required)
- [ ] If making database changes - I have also updated Entity Framework queries and/or migrations
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

**Comment by djsmith85** [comment]
@Manolachi Thank you for your contribution. I've added this to our internal Community PR board for review

File: bitwarden_license/src/Sso/appsettings.Development.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
       "internalAdmin": "http://localhost:62911",
       "internalIdentity": "http://localhost:33656",
       "internalApi": "http://localhost:4000",
-      "internalVault": "http://localhost:4001",
+      "internalVault": "https://localhost:8080",
```
Comment (Hinton) [inline bitwarden_license/src/Sso/appsettings.Development.json:14]: We should revert this change.

File: src/Admin/Controllers/HomeController.cs [hunk 2]
```diff
@@ -60,25 +63,31 @@ public async Task<IActionResult> GetLatestDockerHubVersion(string repository, Ca
                     }
                 }
             }
-            catch (HttpRequestException) { }
+            catch (HttpRequestException e)
+            {
+                _logger.LogError(e, $"Error encountered while sending GET request to {requestUri}");
```
Comment (Hinton) [inline src/Admin/Controllers/HomeController.cs:68]: I wonder if we could just throw a proper exception here which would be picked up in the logs, and return a 500 error which can be handled by the js.

**Review by Hinton — CHANGES_REQUESTED** [review]

**Comment by Hinton** [comment]
Hi @Manolachi,

Thanks for the contribution. I did a quick review, we'll need to revert the changes to the `appsettings`. I also wonder if we could just throw a proper exception in the controller and have it be picked up by the ui. Let me know what your thoughts are.

File: bitwarden_license/src/Sso/appsettings.Development.json (hunk 1 above)
Comment (Hinton) [reply bitwarden_license/src/Sso/appsettings.Development.json:14]: I just saw your comment on the incorrect url. I'll need to investigate if it's used in other places and why it's pointing to 4001.

File: src/Admin/Controllers/HomeController.cs (hunk 2 above)
Comment (Hinton) [reply src/Admin/Controllers/HomeController.cs:68]: @bitwarden/dept-devops Could we potentially return the error message here directly or would that be potentially sensitive information? 

`return new JsonResult(e.Message) { StatusCode = StatusCodes.Status500InternalServerError };`

Comment (joseph-flinn) [reply src/Admin/Controllers/HomeController.cs:68]: It looks like we are using docker hub anonymously, so I don't think there is sensitive information that would be logged.

Comment (Manolachi) [reply src/Admin/Controllers/HomeController.cs:68]: @joseph-flinn Thanks, what about the request on line 79, to the web vault, could that result in an exception with a message that contains sensitive information?

@Hinton Are you thinking of logging the error **and** returning the 500 error, or do you think just returning the 500 error containing the error message is enough?

Comment (Hinton) [reply src/Admin/Controllers/HomeController.cs:68]: @Manolachi Both I think. That way the logs will contain the errors in more details should they be needed.

File: bitwarden_license/src/Sso/appsettings.Development.json (hunk 1 above)
Comment (Hinton) [reply bitwarden_license/src/Sso/appsettings.Development.json:14]: I think we can keep this change.

File: src/Admin/Controllers/HomeController.cs (hunk 2 above)
Comment (Manolachi) [reply src/Admin/Controllers/HomeController.cs:69]: Done.

Comment (Hinton) [reply src/Admin/Controllers/HomeController.cs:69]: Cheers, I'll get this reviewed shortly.

**Comment by Hinton** [comment]
@djsmith85 do you mind reviewing this? I pushed a small change to the error phrasing which ensures it fits into the layout without line breaks 
image

**Review by djsmith85 — COMMENTED** [review]
Looks good to me. 

@Hinton: What about the changes of the appsettings? You mentioned those would need to be reverted?

**Comment by Hinton** [comment]
@djsmith85 We discussed it internally and concluded that the port changes are fine. We haven't used port 4001 in a long time.

**Review by djsmith85 — APPROVED** [review]
Approving this then

**Comment by Hinton** [comment]
Merged, thanks for the contribution @Manolachi.
